### PR TITLE
Remove dependency on FLAGS in TensorFlow utils

### LINKED ISSUE #69: remove all flags from cleverhans module
Having flags (tf flags) is ok in the tutorials, but it would be easier to import functions from the cleverhans module if they didn't reuse flags directly but rather got these values as arguments (e.g., learning rate in model train)

### PR DESCRIPTION
**Pull request by bogdan-kulynych**
Fixes #69
Tested: 

- Ran tests and tutorials to check backwards compatibility
  ```
  tests/test_mnist_accuracy.sh
  tests/test_mnist_accuracy_th.sh
  python tutorials/mnist_tutorial_tf.py
  python tutorials/mnist_tutorial_th.py
   ```
- Modified `tutorials/mnist_tutorial_tf.py`:
  ```
  # - Remove batch_size, nb_epochs, learning_rate flags
  # - Modify model_train, model_eval calls:
  args = {'batch_size': 128, 'nb_epochs': 6, 'learning_rate': 0.1}
  model_train(..., args=args)
  ...
  model_eval(..., args=args)
   ```
-----

Changes:
* Added backwards-compatible argument wrappers
* Added `args` parameter to TF utils functions
* Fixed a bug with mutable default parameter value
* Standardized `args` in TF and theano utils

Note that this produces warnings like this one:

    UserWarning: Setting parameters (batch_size) from TensorFlow FLAGS is deprecated.

I am not sure if the following should go in this PR:

- Add unit tests. Requires setting up unit tests ( #19 ).
- Rewrite TF tutorials and tests to set the args explicitly.
- This may be a good place to change the `args` name both in TF and Theano utils to something like `training_params`

### CODE REVIEW

**Comment by npapernot** [comment]
Thanks for the PR @bogdan-kulynych and the very detailed description!

This is most likely going to break the new tutorial mnist_blackbox.py added last night since it used a similar setup than mnist_tutorial_tf.py. Do you mind pushing the few additional changes required to keep that tutorial alive?

**Comment by bogdan-kulynych** [comment]
Hi @npapernot,
The tutorial seems to be fine. This is what I got:
```
X_train shape: (60000, 28, 28, 1)
60000 train samples
10000 test samples
Preparing the black-box model.
Defined TensorFlow model graph.
/home/bogdan/Workspace/cleverhans/cleverhans/utils_tf.py:35: UserWarning: Setting parameters (learning_rate) from TensorFlow FLAGS is deprecated.
  'deprecated.'.format(name))
/home/bogdan/Workspace/cleverhans/cleverhans/utils_tf.py:35: UserWarning: Setting parameters (nb_epochs) from TensorFlow FLAGS is deprecated.
  'deprecated.'.format(name))
/home/bogdan/Workspace/cleverhans/cleverhans/utils_tf.py:35: UserWarning: Setting parameters (batch_size) from TensorFlow FLAGS is deprecated.
  'deprecated.'.format(name))
Completed model training.
Test accuracy of black-box on legitimate test examples: 0.96969696989
Training the substitute model.
Defined TensorFlow model graph for the substitute.
Substitute training epoch #0
Completed model training.
Augmenting substitute training data.
Labeling substitute training data.
Substitute training epoch #1
Completed model training.
Augmenting substitute training data.
Labeling substitute training data.
Substitute training epoch #2
Completed model training.
Augmenting substitute training data.
Labeling substitute training data.
Substitute training epoch #3
Completed model training.
Augmenting substitute training data.
Labeling substitute training data.
Substitute training epoch #4
Completed model training.
Augmenting substitute training data.
Labeling substitute training data.
Substitute training epoch #5
Completed model training.
Test accuracy of oracle on adversarial examples generated using the substitute: 0.0965656565807
```
The warnings are expected. As I mentioned, I am not sure if rewriting TF tutorials and tests to set the args explicitly should go in this PR. I can do it here though.

**Comment by goodfeli** [comment]
Looks good to me, thanks!
